src/components/encoding-pane/property-editor-schema.ts

```diff
@@ -63,81 +92,217 @@ const POSITION_FIELD_QUANTITATIVE_INDEX = {
-  } else {
-    title = nestedProp || prop;
-  }
+const SIZE_CHANNEL_FIELD_INDEX = {
+  'Common': [
+    {
+      prop: 'legend',
+      nestedProp: 'title'
+    },
+    {
+      prop: 'scale',
+      nestedProp: 'type'
+    }
+  ],
+  'Legend': ['orient', 'title'].map(p => ({prop: 'legend', nestedProp: p})),
+  'Scale': ['domain', 'range', 'type'].map(p => ({prop: 'scale', nestedProp: p}))
+};
 
-  return toTitleCase(title);
-}
+const SHAPE_CHANNEL_FIELD_INDEX = {
+  'Common': [
+    {
+      prop: 'legend',
+      nestedProp: 'orient'
+    },
+    {
+      prop: 'legend',
+      nestedProp: 'title'
+    },
+    {
+      prop: 'scale',
+      nestedProp: 'domain'
+    },
+    {
+      prop: 'scale',
+      nestedProp: 'range'
+    }
+  ]
+};
+
+// ------------------------------------------------------------------------------
+// Color Scheme Constants
+// ------------------------------------------------------------------------------
+export const CATEGORICAL_COLOR_SCHEMES = ['accent', 'category10', 'category20', 'category20b', 'category20c', 'dark2',
+  'paired', 'pastel1', 'pastel1', 'set1', 'set2', 'set3', 'tableau10', 'tableau20'];
+
+export const SEQUENTIAL_COLOR_SCHEMES = ['blues', 'greens', 'greys', 'purples', 'reds', 'oranges', 'viridis', 'inferno',
+  'magma', 'plasma', 'bluegreen', 'bluepurple', 'greenblue', 'orangered', 'blueorange'];
 
-export function getPropertyEditorSchema(prop: string, nestedProp: string, propTab: string): PropertyEditorSchema {
+// ------------------------------------------------------------------------------
+// Generator/Factory Methods
+// ------------------------------------------------------------------------------
+export function generatePropertyEditorSchema(prop: string, nestedProp: string, propTab: string,
+                                             fieldDef: ShelfFieldDef, shelfId: ShelfId): PropertyEditorSchema {
   const title = generateTitle(prop, nestedProp, propTab);
+  const baseUiSchema: SchemaProperty = {
+    title: title,
+    type: 'string'
+  };
+
   if (prop === 'scale') {
     if (nestedProp === 'type') {
+      // Filtering based on channel type & fieldDef type
+      const scaleTypes: string[] = getSupportedScaleTypes(shelfId, fieldDef);
+      const schemaProperty: StringSchema = {
+        ...baseUiSchema,
+        enum: scaleTypes,
+      };
+      return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, schemaProperty);
+    } else if (nestedProp === 'scheme') {
       const schemaProperty: StringSchema = {
-        title: title,
-        enum: (vlSchema as any).definitions.ScaleType.enum,
-        type: 'string'
+        ...baseUiSchema,
+        enum: isContinuous(fieldDef) ? SEQUENTIAL_COLOR_SCHEMES : CATEGORICAL_COLOR_SCHEMES,
       };
       return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, schemaProperty);
+    } else if (nestedProp === 'range' || nestedProp === 'domain') {
+      const schemaProperty: StringSchema = {
```

Comment:
1) why do you need to do ... at all, can't you just use baseUISchema
2) baseUISchema, not baseUiSchema
3) I just notice this is a property's schema, not a schema's property.   Better name all of them `schemaProperty`?